packages/kit/src/runtime/server/cookie.js

```diff
@@ -0,0 +1,76 @@
+import * as cookie from 'cookie';
+
+/**
+ * @param {Request} request
+ * @param {URL} url
+ */
+export function get_cookies(request, url) {
+	const initial_cookies = cookie.parse(request.headers.get('cookie') ?? '');
+
+	/** @type {Array<{ name: string, value: string, options: import('cookie').CookieSerializeOptions }>} */
+	const new_cookies = [];
+
+	/** @type {import('types').Cookies} */
+	const cookies = {
+		get(name, opts) {
+			const decode = opts?.decode || decodeURIComponent;
+
+			let i = new_cookies.length;
+			while (i--) {
+				const cookie = new_cookies[i];
+
+				if (
+					cookie.name === name &&
+					domain_matches(url.hostname, cookie.options.domain) &&
+					path_matches(url.pathname, cookie.options.path)
+				) {
+					return cookie.value;
+				}
+			}
+
+			return decode(initial_cookies[name]);
```

Comment:
I think you want to check for the cookie's existence first, since `decodeURIComponent(undefined)` gives the string `'undefined'` not `undefined`.

```ts
return typeof initial_cookies[name] === 'string' ? decode(initial_cookies[name]) : undefined;
```

Reply:
ah, good catch. very slightly leaner:

```suggestion
			return name in initial_cookies ? decode(initial_cookies[name]) : undefined;
```